src/utils/file_upload.rs

```diff
@@ -252,6 +260,10 @@ fn upload_files_chunked(
             break response;
         }
 
+        if context.wait && assemble_start.elapsed() > max_wait {
```

Comment:
```suggestion
        if assemble_start.elapsed() > max_wait {
```

`context.wait` will always evaluate to `true` here, as L259 would short-circuit it already if it'd be `false`